linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/LoaderContent.scala

```diff
@@ -157,6 +157,13 @@ const scalaJSHelpers = {
   jsNewNoArg: (constr) => new constr(),
   jsImportCall: (s) => import(s),
   jsImportMeta: () => import.meta,
+  jsAwait: (WebAssembly.Suspending ? new WebAssembly.Suspending((x) => x) : ((x) => {
```

Comment:
Can we do this statically based on linking info rather than auto-detect?

- Less runtime magic
- Less code
- More eager failure

Reply:
We would have to add that information in the `Analysis`. Currently it doesn't have it.

My idea here was rather that eventually we'll remove the fallback altogether. JSPI will become part of standard Wasm-on-JS, so every JS host will have `WebAssembly.Suspending`. So spending extra effort to track something we eventually won't use at all is probably not worth it.